__tests__/app.test.js

```diff
@@ -40,65 +40,106 @@ describe("API Testing", () => {
+					.expect(200)
+					.then(({ body }) => {
+						const article = body.article;
+						expect(article).toBeInstanceOf(Array);
+						expect(article.length).not.toBeLessThan(1);
+						article.forEach((article) => {
+							expect(article.article_id).toBe(3);
+							expect(article).toHaveProperty("author");
+							expect(article).toHaveProperty("title");
+							expect(article).toHaveProperty("article_id");
+							expect(article).toHaveProperty("body");
+							expect(article).toHaveProperty("topic");
+							expect(article).toHaveProperty("created_at");
+							expect(article).toHaveProperty("votes");
+							expect(article).toHaveProperty("article_img_url");
+						});
 					});
-				});
-		});
-		test("Err: 404, Returns error 404 when given a valid but non existant article is", () => {
-			return request(app)
-				.get("/api/articles/200")
-				.expect(404)
-				.then((response) => {
-					expect(response.body.msg).toBe("article does not exist");
-				});
+			});
+			test("Err: 404, Returns error 404 when given a valid but non existant article is", () => {
+				return request(app)
+					.get("/api/articles/200")
+					.expect(404)
+					.then((response) => {
+						expect(response.body.msg).toBe("article does not exist");
+					});
+			});
+			test("Err: 400, should be able to handle a get request when there are no matches with valid url", () => {
+				return request(app)
+					.get("/api/articles/article")
+					.expect(400)
+					.then((response) => {
+						expect(response.body.msg).toBe("Bad request");
+					});
+			});
 		});
-		test("Err: 400, should be able to handle a get request when there are no matches with valid url", () => {
-			return request(app)
-				.get("/api/articles/article")
-				.expect(400)
-				.then((response) => {
-					expect(response.body.msg).toBe("Bad request");
-				});
+		describe("Get comment for article", () => {
+			test("Get: 200,  should return all comments for an article", () => {
+				return request(app)
+					.get("/api/articles/1/comments")
+					.expect(200)
+					.then(({ body }) => {
+						const comments = body.comments;
+						expect(comments).toBeInstanceOf(Array);
+						expect(comments.length).not.toBeLessThan(1);
+						expect(comments).toBeSortedBy("created_at", { descending: true });
+						comments.forEach((comment) => {
+							expect(comment.article_id).toBe(1);
+							expect(comment).toHaveProperty("comment_id");
+							expect(comment).toHaveProperty("votes");
+							expect(comment).toHaveProperty("created_at");
+							expect(comment).toHaveProperty("author");
+							expect(comment).toHaveProperty("body");
+							expect(comment).toHaveProperty("article_id");
+						});
+					});
+			});
+			test("Err: 404, Returns error 404 when given a valid but non existant article is", () => {
+				return request(app)
+					.get("/api/articles/2/comments")
+					.expect(404)
+					.then((response) => {
+						expect(response.body.msg).toBe("article has no comments");
+					});
+			});
```

Comment:
Should this be a 404? Or should it be a 200 with an empty array which says there's no comments currently... The request shouldn't fail or have an error, it would just be that the article_id exists, with no current comments.

The 404 test would however be appropriate for an endpoint such as: `/api/articles/3763322/comments` - a valid article id which does not exist in the database.